vispy/visuals/graphs/util.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""
+Graph utilities
+===============
+
+A module containing several graph utility functions.
+"""
+
+import itertools
+
+import numpy as np
+
+try:
+    from scipy.sparse import issparse
+    from scipy import sparse
+except ImportError:
+    def issparse(*args, **kwargs):
+        return False
+
+
+def _get_edges(adjacency_mat):
+    func = _sparse_get_edges if issparse(adjacency_mat) else _ndarray_get_edges
+    return func(adjacency_mat)
+
+
+def _sparse_get_edges(adjacency_mat):
+    return np.concatenate((adjacency_mat.row[:, np.newaxis],
+                           adjacency_mat.col[:, np.newaxis]), axis=-1)
+
+
+def _ndarray_get_edges(adjacency_mat):
+    # Get indices of all non zero values
+    i, j = np.where(adjacency_mat)
+
+    return np.concatenate((i[:, np.newaxis], j[:, np.newaxis]), axis=-1)
+
+
+def _get_directed_edges(adjacency_mat):
+    func = _sparse_get_edges if issparse(adjacency_mat) else _ndarray_get_edges
+    triu = sparse.triu if issparse(adjacency_mat) else np.triu
+    tril = sparse.tril if issparse(adjacency_mat) else np.tril
```

Comment:
once you have more than one (or maybe two) of the same conditional, you should probably split it up:
```
if issparse(adjacency_mat):
   ...
else:
   ...
```